server/src/test/java/com/objectcomputing/checkins/services/pulseResponse/PulseResponseControllerTest.java

```diff
@@ -0,0 +1,256 @@
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.micronaut.test.annotation.MicronautTest;
+
+@TestInstance(Lifecycle.PER_CLASS)
+@MicronautTest
+public class PulseResponseControllerTest {
+    
+    @Inject
+    @Client("/pulse-response")
+    private HttpClient client;
+
+    @Inject
+    MemberProfileController memberProfileController;
+    
+    PulseResponseRepository mockPulseResponseRepository = mock(PulseResponseRepository.class);
+    PulseResponse mockPulseResponse = mock(PulseResponse.class);
+    
+    private static LocalDate testsubmissionDate = LocalDate.of(2020, 1, 01);
+    private static LocalDate testUpdatedDate = LocalDate.of(2020, 2, 01);
+    private static UUID testTeamMemberId;
+    private static boolean isDataSetupForTest = false;
+    
+    @Before
+    void setup() {
+        reset(mockPulseResponseRepository);
+        reset(mockPulseResponse);
+    }
+    
+    @BeforeAll
+    void setupMemberProfileRecord() {
+        // setup a record in Member-Profile to satisfy foreign key constraint
+        final Date testDate = new Date(System.currentTimeMillis());
+
+        if(memberProfileController != null) {
+            MemberProfile testMemberProfile = new MemberProfile("TestName", 
+                                                                "TestRole", 
+                                                                UUID.randomUUID(), 
+                                                                "TestLocation", 
+                                                                "TestEmail", 
+                                                                "TestInsperityId", 
+                                                                testDate, 
+                                                                "TestBio");
+
+            final HttpResponse<?> response = memberProfileController.save(testMemberProfile);
+            assertEquals(HttpStatus.CREATED, response.getStatus());
+            testTeamMemberId = ((MemberProfile) response.body()).getUuid();
+        }
+    }
+
+    @Test
+    public void testFindNonExistingEndpointReturns404() {
+        HttpClientResponseException thrown = assertThrows(HttpClientResponseException.class, () -> {
+            client.toBlocking().exchange(HttpRequest.GET("/99"));
+        });
+
+        assertNotNull(thrown.getResponse());
+        assertEquals(HttpStatus.NOT_FOUND, thrown.getStatus());
+    }
+
+    @Test
+    public void testFindNonExistingEndpointReturnsNotFound() {
+        HttpClientResponseException thrown = assertThrows(HttpClientResponseException.class, () -> {
+            client.toBlocking().exchange(HttpRequest.GET("/bar?order=foo"));
+        });
+
+        assertNotNull(thrown.getResponse());
+        assertEquals(HttpStatus.NOT_FOUND, thrown.getStatus());
+    }
```

Comment:
How are the two tests above different?

Reply:
They weren't different from one another. I've removed one of them.